crates/hir-ty/src/infer.rs

```diff
@@ -1152,20 +1152,14 @@ impl<'a> InferenceContext<'a> {
                 (ty, variant)
             }
             TypeNs::TypeAliasId(it) => {
-                let container = it.lookup(self.db.upcast()).container;
-                let parent_subst = match container {
-                    ItemContainerId::TraitId(id) => {
```

Comment:
It doesn't seem possible to encounter this -- trait assoc type would either results in Self or Trait.